src/app/app.datatypes.ts

```diff
@@ -63,14 +64,32 @@ export class GetAddressResponseTransaction {
   txid: string;
 }
 
-export function parseGetAddressTransaction(raw: GetAddressResponseTransaction): Transaction {
+export function parseGetAddressTransaction(raw: GetAddressResponseTransaction, address: string): Transaction {
+
+  //Detect if the address sent or received the coins.
+  let incoming: boolean = true;
```

Comment:
You don't have to type-cast it as a boolean, typescript will auto typecast as the default value is true